src/http_lwt_client.ml

```diff
@@ -268,7 +267,9 @@ let single_h2_request ?config fd scheme user_pass host meth path headers body f
    | Some body -> H2.Body.Writer.write_string request_body body
    | None -> ());
   H2.Body.Writer.close request_body;
-  finished
+  finished >|= fun res ->
```

Comment:
We're using the `Lwt_result.Infix` operators in this module -- so I wonder whether the call to `shutdown` below should be done in all cases (not only the successful ones)?

Reply:
I pushed a commit that handles if `finished` is somehow cancelled using `Lwt.finalize`. Otherwise `finished` returns a result so I think it handles failures already? What do you have in mind?

Reply:
Looks fine, the alternative would be:

```OCaml
Lwt.bind finished (fun r -> H2.Client_connection.shutdown connection; Lwt.return r)
```

I suspect that `finalize` is preferably - and still suspect that "cancelling http_lwt_client request" may lead to resource leaks... (but that's fine since it was the case before)